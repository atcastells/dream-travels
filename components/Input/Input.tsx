import classNames from "classnames";
import { DetailedHTMLProps, InputHTMLAttributes } from "react";

export type InputProps = DetailedHTMLProps<
  InputHTMLAttributes<HTMLInputElement>,
  HTMLInputElement
> & {
  label?: string;
  error?: string;
  containerProps?: DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  >;
};

export const Input = ({
  error,
  label,
  containerProps,
  ...inputProps
}: InputProps) => {
  return (
    <div
      {...containerProps}
      className={classNames(
        "flex flex-col text-sm rounded-full",
        containerProps?.className ? containerProps.className : ""
      )}
    >
      {label && <label className="text-sm mb-1">{label}</label>}

      <div
        className={classNames(
          "flex rounded-full border bg-white border-gray-200 p-1",
          {
            "border-red-400": !!error,
          }
        )}
      >
        <input
          {...inputProps}
          type="text"
          className={classNames(
            "rounded-full bg-white text-black px-2 outline-none w-full",
            inputProps.className ? inputProps.className : ""
          )}
        />
      </div>
      {error && <div className="text-red-400 text-xs mt-2 ml-4">{error}</div>}
    </div>
  );
};
