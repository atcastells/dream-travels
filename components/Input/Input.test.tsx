import { expect, test } from "@jest/globals";
import { render, screen } from "@testing-library/react";

import { Input } from "./Input";

test("renders input", () => {
  render(<Input />);
  const input = screen.getByRole("textbox");
  expect(input).toBeTruthy();

  const label = screen.queryByRole("label");
  expect(label).toBeNull();

  const error = screen.queryByText("error");
  expect(error).toBeNull();

  expect(input).toMatchSnapshot();
});
