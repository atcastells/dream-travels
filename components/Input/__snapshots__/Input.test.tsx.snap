// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders input 1`] = `
<input
  class="rounded-full bg-white text-black px-2 outline-none w-full"
  type="text"
/>
`;
